apps/api/src/server/applications/sector/sector.controller.js

```diff
@@ -21,12 +22,18 @@ const mapSectors = (sectors) => {
  * @type {import('express').RequestHandler}
  */
 export const getSectors = async (request, response) => {
-	const sectors = isEmpty(request.query)
-		? await sectorRepository.getAll()
-		: await subSectorRepository.getBySector({ name: request.query.sectorName?.toString() });
+	let sectors = await getCache(request.query.sectorName?.toString() || '');
 
-	// sort ascending order of subsector abbreviation, BC, EN, ... WA, WS, WW
-	sectors.sort((a, b) => (a.abbreviation > b.abbreviation ? 1 : -1));
+	if (!sectors) {
+		sectors = isEmpty(request.query)
+			? await sectorRepository.getAll()
+			: await subSectorRepository.getBySector({ name: request.query.sectorName?.toString() });
+
+		// sort ascending order of subsector abbreviation, BC, EN, ... WA, WS, WW
+		sectors.sort((a, b) => (a.abbreviation > b.abbreviation ? 1 : -1));
```

Comment:
Something that we should be doing is sorting in the database, so let's update the call to the database such that we receive a sorted list back

Reply:
I added the sort fx to a file in the utils folder - called it 'sortAll' - then I wrap that fx around 'getAll' in sectorRepository, and 'getBySector' in subSectorRepository -- is this correct?

Reply:
So what I mean is that there are generally two spaces we can sort things: in the memory of the application itself or in the database. Ideally, we want to do all sorting in the database, so the way we can do that is by send to prisma that we would like the items sorted. So moving the code to a different space is not going to help us here